jerry-libc/jerry-libc.c

```diff
@@ -27,7 +27,11 @@
 /**
  * State of pseudo-random number generator
  */
-static uint32_t libc_random_gen_state[4] = { 1455997910, 1999515274, 1234451287, 1949149569 };
+static uint32_t initial_states[3] = { 842502087, 3579807591, 273326509};
+static uint32_t _x;
+static uint32_t _y;
+static uint32_t _z;
+static uint32_t _w;
```

Comment:
The patch identifies correctly that `srand` should not set all state elements to the same value. However, that does not mean that the whole RNG, together with its state representation and `rand` implementation should be rewritten (there is no problem with any of those).

Moreover, this change here does not only rewrite the state representation but introduces a new issue by leaving `_x`, `_y`, `_z`, and `_w` uninitialized.

Please, revert this block to the original code version (with `libc_random_gen_state`).